### PR TITLE
Prepare for Release Kirkstone 1.5 LTS

### PR DESCRIPTION
**Pull request by ggjjj**
Kikstone release for 1.5 LTS for IoT Edge. Recipes are updated and new patch files are added to accommodate the changes in IoT Edge and IoT Identity Service. Going forward we will also replace the bitbake files instead of keeping all the prev versions.

### CODE REVIEW

File: recipes-core/aziot-edged/aziot-edged_1.5.5.bb [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=http-common;destsuffix=http-common;subpath=http-common \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=logger;destsuffix=logger;subpath=logger \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=openssl-build;destsuffix=openssl-build;subpath=openssl-build \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=openssl-sys2;destsuffix=openssl-sys2;subpath=openssl-sys2 \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=openssl2;destsuffix=openssl2;subpath=openssl2 \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=pkcs11-sys;destsuffix=pkcs11/pkcs11-sys;subpath=pkcs11/pkcs11-sys \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=pkcs11;destsuffix=pkcs11/pkcs11;subpath=pkcs11/pkcs11 \
+    git://github.com/Azure/iot-identity-service;protocol=https;nobranch=1;name=test-common;destsuffix=test-common;subpath=test-common \
```
Comment (damonbarry) [inline recipes-core/aziot-edged/aziot-edged_1.5.5.bb:273]: In past releases we deleted these autogenerated github URIs. Do we not need to do that anymore?

File: recipes-core/aziot-edged/aziot-edged_1.5.5.bb [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+EXTRA_OECARGO_PATHS += "${WORKDIR}/openssl2"
+SRCREV_FORMAT .= "_pkcs11"
+SRCREV_pkcs11 = "b9fff6b2fdf9c1593a5a0b7e856a5f01c2c5ad5b"
+EXTRA_OECARGO_PATHS += "${WORKDIR}/pkcs11/pkcs11"
+SRCREV_FORMAT .= "_pkcs11-sys"
+SRCREV_pkcs11-sys = "b9fff6b2fdf9c1593a5a0b7e856a5f01c2c5ad5b"
+EXTRA_OECARGO_PATHS += "${WORKDIR}/pkcs11/pkcs11-sys"
+SRCREV_FORMAT .= "_test-common"
+SRCREV_test-common = "b9fff6b2fdf9c1593a5a0b7e856a5f01c2c5ad5b"
+EXTRA_OECARGO_PATHS += "${WORKDIR}/test-common"
```
Comment (damonbarry) [inline recipes-core/aziot-edged/aziot-edged_1.5.5.bb:579]: Same as previous comment...we used to delete these autogenerated SRCREV_FORMAT, SRCREV_*, and EXTRA_OECARGO_PATHS variable assignments from the .bb file. Do we not need to do that anymore? The less changes we need to make the better IMO, but I wanted to double check. :)

File: scripts/build.sh [hunk 3]
```diff
@@ -17,5 +17,5 @@ die() {
 rm -f build/conf/bblayers.conf || die "failed to nuke bblayers.conf"
 rm -f build/conf/local.conf || die "failed to nuke local.conf"
 
-./scripts/containerize.sh ./scripts/bitbake.sh ${BUILD_TARGETS} || die "failed to build"
+./scripts/containerize.sh  ./scripts/bitbake.sh ${BUILD_TARGETS} || die "failed to build"
```
Comment (damonbarry) [inline scripts/build.sh:20]: Is the extra space needed here?

**Comment by Dvergatal** [comment]
I'm still getting error related to `frozen` param in `openembedded-core/meta/classes-recipe/cargo.bbclass` looking like:

```bash
ERROR: iotedge-1.5.5-r0 do_compile: ExecutionError('/work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317', 101, None, None)
ERROR: Logfile of failure stored in: /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/log.do_compile.4036317
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: Using rust targets from /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/rust-targets/
| NOTE: cargo = /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/recipe-sysroot-native/usr/bin/cargo
| NOTE: cargo build -v --frozen --target x86_64-welotec-linux-gnu --release --manifest-path=/work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/git/edgelet/iotedge/Cargo.toml
| error: the lock file /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/git/edgelet/Cargo.lock needs to be updated but --frozen was passed to prevent this
| If you want to try to generate the lock file without accessing the network, remove the --frozen flag and use --offline instead.
| WARNING: /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317:195 exit 101 from '"cargo" build -v --frozen --target x86_64-welotec-linux-gnu --release --manifest-path=/work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/git/edgelet/iotedge/Cargo.toml "$@"'
| WARNING: Backtrace (BB generated script):
|       #1: oe_cargo_build, /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317, line 195
|       #2: cargo_do_compile, /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317, line 167
|       #3: do_compile, /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317, line 162
|       #4: main, /work/build/tmp-glibc/work/core2-64-welotec-linux/iotedge/1.5.5/temp/run.do_compile.4036317, line 208
ERROR: Task (/work/build/yocto/meta-iotedge/recipes-core/iotedge/iotedge_1.5.5.bb:do_compile) failed with exit code '1'
```

File: scripts/build.sh (hunk 3 above)
Comment (ggjjj) [reply scripts/build.sh]: done

**Review by damonbarry — APPROVED** [review]

**Comment by damonbarry** [comment]
The release on which this PR is based--IoT Edge 1.5.5--is a bit outdated. But it's fine with me if you want to get this in as a baseline. We can do a follow up to sync to the latest release.

File: recipes-core/aziot-edged/aziot-edged_1.5.5.bb (hunk 1 above)
Comment (ggjjj) [reply recipes-core/aziot-edged/aziot-edged_1.5.5.bb:273]: We need it to lock down a specific version of crates for now. In future releases will find a way without this

**Pull request merged**